athena-elasticsearch/src/main/java/com/amazonaws/connectors/athena/elasticsearch/ElasticsearchTypeUtils.java

```diff
@@ -0,0 +1,539 @@
+     * as a String or an ArrayList. For the latter, extract the first element only.
+     * @param field is used to determine which extractor to generate based on the field type.
+     * @return a field extractor.
+     */
+    private Extractor makeVarCharExtractor(Field field)
+    {
+        return (VarCharExtractor) (Object context, NullableVarCharHolder dst) ->
+        {
+            Object fieldValue = ((Map) context).get(field.getName());
+            dst.isSet = 1;
+            if (fieldValue instanceof String) {
+                dst.value = (String) fieldValue;
+            }
+            else if (fieldValue instanceof ArrayList) {
+                Object value = ((ArrayList) fieldValue).get(0);
+                if (value instanceof String) {
+                    dst.value = (String) value;
+                }
+                else {
+                    dst.isSet = 0;
+                }
+            }
+            else {
+                dst.isSet = 0;
+            }
+        };
+    }
+
+    /**
+     * Create a BIGINT field extractor to extract a long value from a Document. The Document value can be returned
+     * as a numeric value, a String, or an ArrayList. For the latter, extract the first element only. Special logic
+     * is employed to parse out a scaled_float which is a float represented as a long (scaled by a double value).
+     * @param field is used to determine which extractor to generate based on the field type.
+     * @return a field extractor.
+     */
+    private Extractor makeBigIntExtractor(Field field)
+    {
+        return (BigIntExtractor) (Object context, NullableBigIntHolder dst) ->
+        {
+            Object fieldValue = ((Map) context).get(field.getName());
+            dst.isSet = 1;
+            if (!field.getMetadata().isEmpty() && field.getMetadata().containsKey("scaling_factor")) {
+                // scaled_float w/scaling_factor - a float represented as a long.
+                double scalingFactor = new Double(field.getMetadata().get("scaling_factor"));
+                if (fieldValue instanceof Number) {
+                    dst.value =  Math.round(((Number) fieldValue).doubleValue() * scalingFactor);
+                }
+                else if (fieldValue instanceof String) {
+                    dst.value = Math.round(new Double((String) fieldValue) * scalingFactor);
+                }
+                else if (fieldValue instanceof ArrayList) {
+                    Object value = ((ArrayList) fieldValue).get(0);
+                    if (value instanceof Number) {
+                        dst.value =  Math.round(((Number) value).doubleValue() * scalingFactor);
+                    }
+                    else if (value instanceof String) {
+                        dst.value = Math.round(new Double((String) value) * scalingFactor);
+                    }
+                    else {
+                        dst.isSet = 0;
```

Comment:
anywhere you find yourself doing dst.isSet = 0, please ensure this is the right behavior vs throwing because you failed to coerce a real value. dst.isSet should only be 0 if there is no value in the source and in such cases I don't think this method would even be called. you can confirm with a test by ommitting the field from some of your documents in elastic.

Reply:
this is an indication that the field returned a null value (e.g. "field": null). In Elasticsearch the fields will be returned as the following class instances: String, Number, Boolean, ArrayList, HashMap,  and **null**.